leshan-server-core/src/main/java/org/eclipse/leshan/server/Destroyable.java

```diff
@@ -15,6 +15,10 @@
  *******************************************************************************/
 package org.eclipse.leshan.server;
 
+/**
+ * @deprecated please consider to use {@link org.eclipse.leshan.core.Destroyable} instead.
+ */
+@Deprecated
```

Comment:
Everywhere we use the old  `Destroyable` interface we should now be able to use the new one.
(I created a new PR with this modification : df5725cc3410e2b771989b98ac6e11a995f0dc53)